x/plans/keeper/plan.go

```diff
@@ -23,6 +23,18 @@ func (k Keeper) AddPlan(ctx sdk.Context, planToAdd types.Plan) error {
 	return nil
 }
 
+// DelPlan deletes a plan, so it is not visible/gettable for new subscriptions
+// (however, existing referenced versions remain intact until not used anymore)
+func (k Keeper) DelPlan(ctx sdk.Context, index string) error {
+	// Deletions should take place at the end of epoch (beginning of next epoch).
+	// However, because deletion of plans occurs through gov proposal, we must be
+	// already at the beginning of that "next" epoch (after having waited for the
+	// already at the beginning of that "next" epoch (after having waited for the
+	// the voting). Hence, we delete at current block height.
+
+	return k.plansFS.DelEntry(ctx, index, uint64(ctx.BlockHeight()))
```

Comment:
we want all effects to be at the start of an epoch, proposals are not always effective at that time, we should delete the proposal at the start of the next epoch

Reply:
ok, fixed.